aiida_vasp/calcs/neb.py

```diff
@@ -0,0 +1,313 @@
+            for name in included:
+                if name not in existing_files:
+                    # Here we simple issue an warning as the requirement of files will be explicitly checked by
+                    # `write_additional` method
+                    self.report('WARNING: File {} does not exist in the restart folder.'.format(name))
+                else:
+                    copy_list.append((
+                        computer.uuid,
+                        os.path.join(restart_folder.get_remote_path(), fdname, name),
+                        os.path.join(fdname, name),
+                    ))
+
+        return copy_list
+
+    def write_neb_poscar(self, structure, dst, positions_dof=None):  # pylint: disable=unused-argument
+        """
+        Write the POSCAR.
+
+        Passes the structures node (StructureData) to the POSCAR parser for
+        preparation and writes to dst.
+
+        :param dst: absolute path of the file to write to
+        """
+        settings = self.inputs.get('settings')
+        settings = settings.get_dict() if settings else {}
+        poscar_precision = settings.get('poscar_precision', 10)
+        if positions_dof is not None:
+            options = {'positions_dof': positions_dof}
+        else:
+            options = None
+        poscar_parser = PoscarParser(data=self._structure(), precision=poscar_precision, options=options)
+        poscar_parser.write(dst)
+
+    def _structure(self):
+        """
+        Get the input structure as AiiDa StructureData.
+
+        This is required in order to support CifData as input as well.
+        """
+        structure = self.inputs.initial_structure
+        if not hasattr(structure, 'get_pymatgen'):
+            structure = get_data_node('structure', ase=structure.get_ase())
+        return structure
+
+    def verify_inputs(self):
+        """
+        Verify the order of elements
+        """
+        super().verify_inputs()
+        last_order = None
+        last_num_sites = None
+        for structure in list(self.inputs.neb_images.values()) + [self.inputs.initial_structure, self.inputs.final_structure]:
+            # Convert to StructureData from CifData on demand....
+            if not hasattr(structure, 'get_pymatgen'):
+                structure = get_data_node('structure', ase=structure.get_ase())
+
+            num_sites = len(structure.sites)
+
+            order = ordered_unique_symbols(structure)
+            if last_order is not None and order != last_order:
+                raise InputValidationError('The input structures have non-euqal element orders.')
+            if last_order is not None and num_sites != last_num_sites:
+                raise InputValidationError('The input structures have non-equal number of atoms.')
+            last_order = order
+            last_num_sites = num_sites
+
+    def write_incar(self, dst, validate_tags=False):  # pylint: disable=unused-argument
```

Comment:
I guess we could reuse the `write_incar` from `VaspCalculation`, but we would need to set the `validate_tags`. Also, this should be added to `parsevasp`. I opened an issue https://github.com/aiida-vasp/parsevasp/issues/105. But let us leave it for this PR.